lua/starfall/libs_cl/render.lua

```diff
@@ -748,6 +760,180 @@ function render_library.setMaterial(mat)
+	pp.bloom:SetFloat("$levelg", levelg)
+	pp.bloom:SetFloat("$levelb", levelb)
+	pp.bloom:SetFloat("$colormul", colormul)
+	surface.SetMaterial(pp.bloom)
+	render.SetMaterial(pp.bloom)
+
+end
+
+--- Sets the current render material to the given material or the rendertarget, darkening the texture, and scaling up color values.
+-- @param mat The material object to use the texture of, or the name of a rendertarget to use instead.
+-- @param darken The amount to darken the texture by. -1 to 1 inclusive.
+-- @param multiply The amount to multiply the pixel colors by.
+function render_library.setMaterialEffectDownsample(mat, darken, multiply)
+
+	checkpermission(SF.instance, nil, "render.effects")
+	local data = SF.instance.data.render
+	if not data.isRendering then SF.Throw("Not in rendering hook.", 2) end
+	local tex = gettexture(mat)
+	checkluatype(darken, TYPE_NUMBER)
+	checkluatype(multiply, TYPE_NUMBER)
+	darken = math.Clamp(darken, -1, 1)
+	multiply = math.Clamp(multiply, 0, 1024)
+
+	pp.downsample:SetTexture("$fbtexture", tex)
+	pp.downsample:SetFloat("$darken", darken)
+	pp.downsample:SetFloat("$multiply", multiply)
+	surface.SetMaterial(pp.downsample)
+	render.SetMaterial(pp.downsample)
+
+end
+
+
+local defaultCM = {
+	addr = 0,
+	addg = 0,
+	addb = 0,
+	brightness = 0,
+	colour = 1,
+	contrast = 1,
+	mulr = 1,
+	mulg = 1,
+	mulb = 1
+}
+
+--- Sets the current render material to the given material or the rendertarget, applying a color modification shader to the texture. Alias: render.setMaterialEffectColourModify
+-- @param mat The material object to use the texture of, or the name of a rendertarget to use instead.
+-- @param cmStructure A table where each key must be of "addr", "addg", "addb", "brightness", "color" or "colour", "contrast", "mulr", "mulg", and "mulb". All keys are optional.
+function render_library.setMaterialEffectColorModify(mat, cmStructure)
+
+	checkpermission(SF.instance, nil, "render.effects")
+	local data = SF.instance.data.render
+	if not data.isRendering then SF.Throw("Not in rendering hook.", 2) end
+	local tex = gettexture(mat)
+	checkluatype(cmStructure, TYPE_TABLE)
+
+	for key, default in pairs(defaultCM) do
+		local value = cmStructure[key]
+		if TypeID(value) == TYPE_NIL then
+			if key == "colour" then
+				value = cmStructure["color"] or default
+			else
+				value = default
+			end
+		elseif TypeID(value) ~= TYPE_NUMBER then SF.Throw("Invalid type for key \"" .. key .. "\" (expected number, got " .. SF.GetType(value) .. ")", 2) end
+
+		value = math.Clamp(value, -1024, 1024)
+		pp.colour:SetFloat("$pp_colour_" .. key, value)
+	end
+
+	pp.colour:SetTexture("$fbtexture", tex)
+	surface.SetMaterial(pp.colour)
+	render.SetMaterial(pp.colour)
+
+end
+
+render_library.setMaterialEffectColourModify = render_library.setMaterialEffectColorModify
+
+
+local aspectRatio = ScrW() / ScrH() -- For some reason, render.BlurRenderTarget performs the blur based on the game's window size. This compensates for that
```

Comment:
If the user changes their resolution while in-game this could become inaccurate.

Reply:
I'll go ahead and add a fix for this, real quick